Original patch from: ../4.3.2/350-sh-without-headers.patch

-= BEGIN original header =-
On 20081011, Khem RAJ writes:
 You have to bar gcc trying to build unwinding at stage 1. Try the
 attached patch and see if it helps.

 gcc/config/sh/linux-unwind.h |    3 +++
 1 file changed, 3 insertions(+)

-= END original header =-

diff -durN gcc-4.3.3.orig/gcc/config/sh/linux-unwind.h gcc-4.3.3/gcc/config/sh/linux-unwind.h
--- gcc-4.3.3.orig/gcc/config/sh/linux-unwind.h	2007-09-12 08:59:06.000000000 +0200
+++ gcc-4.3.3/gcc/config/sh/linux-unwind.h	2009-01-27 22:28:57.000000000 +0100
@@ -26,6 +26,8 @@
 the Free Software Foundation, 51 Franklin Street, Fifth Floor,
 Boston, MA 02110-1301, USA.  */
 
+#ifndef inhibit_libc
+
 /* Do code reading to identify a signal frame, and set the frame
    state data appropriately.  See unwind-dw2.c for the structs.  */
 
@@ -251,3 +253,4 @@
   return _URC_NO_REASON;
 }
 #endif /* defined (__SH5__) */
+#endif /* ifndef inhibit_libc */
